### DISCUSSION TITLE
v7 to v8 codemod

### DISCUSSION THREAD

**Discussion by jrdavismaine** [webdriverio/codemod#31] [General]
Responding to "more will eventually follow, let us know which ones you would like to see" from the README.

I am hoping a v7 to v8 codemod will be created to help transition any sync tests to async/await. Looking forward to an async only WebdriverIO. Thank you for creating such powerful automation framework.

**Comment by christian-bromann** [comment]
@jdavis61 once we release v8 we definitely make sure to have the proper migration path for users ready. Thanks for logging this issue.

**Reply by jrdavismaine** [reply]
Awesome! :) Looking forward to V8 and transferring all of my sync tests to async.  Would any future v7 to v8 codemod focus on just converting the webdriverio commands and not the test framework / page objects? 

For example, my assumption is the v7 to v8 coedmod would convert:
```
public loadNavigationMenu() : void {
    browser.url('/');
}

public loadFormAuthenticationPage() : void {
    browser.url('/login');
    browser.$(selectors.screenPresenceLocators.login).waitForDisplayed();
}
```

to something like this:

```
public loadNavigationMenu() : void {
   await browser.url('/');
}

public loadFormAuthenticationPage() : void {
    await browser.url('/login');
    await browser.$(selectors.screenPresenceLocators.login).waitForDisplayed();
}
```

The v7 to v8 codemod **would not** add `async` before the `public` modifiers though, e.g.: 
```
async public loadNavigationMenu() : void {
   await browser.url('/');
}
```
Another codemod would have to do this. Also if running these functions in mocha we would have to rely on a mocha codemod, e.g. the codemod would not convert: 
```
it('Run test', () => {
  loadNavigationMenu();
  loadFormAuthenticationPage();
});
```

to 

```
it('Run test', async () => {
  await loadNavigationMenu();
  await loadFormAuthenticationPage();
});
```
Anyways I realize V8 is not done yet and probably will not be for a while. More thinking out loud, trying to understand how this will work.  Cheers!

**Reply by christian-bromann** [reply]
> to something like this:

Yes, all statements that have a call expression that contains a property with a WebdriverIO command can be easily transformed, some of the experiments I did with a Babel transpiler worked very well.

> the codemod would not convert:

Correct, this is the reason why we stopped with the idea of transforming the test code ourselves from sync to async and allow users to continue write tests with sync code. Such function calls need to be transformed manually but the codemod can help discover them.
